src/DIRAC/WorkloadManagementSystem/Service/JobStateUpdateHandler.py

```diff
@@ -282,6 +282,11 @@ def _setJobStatusBulk(cls, jobID, statusDict, force=False):
             )
             if not result["OK"]:
                 return result
+            # If the update comes from a job, update the heart beat time stamp
+            if source.startswith("Job"):
```

Comment:
How big of a dictionary is `updateTimes` (12 lines up), and would it make sense to move this update outside the loop if any source starts with "Job"?